DeratControl.Application/Points/Queries/GetPointsByPerimeter/GetPointsQueryHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using DeratControl.Application.Interfaces;
+using DeratControl.Application.Requests;
+using DeratControl.Application.Requests.Interfaces;
+using DeratControl.Domain.Entities;
+using DeratControl.Domain.Root;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DeratControl.Application.Points.Queries.GetPointsByPerimeter
+{
+    public class GetPointsQueryHandler : IQueryHandler<GetPointsQuery,PointsViewModelResult>
+    {
+        private IUnitOfWork _unitofwork;
+
+        public GetPointsQueryHandler(IUnitOfWork unitofwork)
+        {
+            this._unitofwork = unitofwork;
+        }
+
+       public Task<PointsViewModelResult> Handle(CommandExecutionContext executionContext, GetPointsQuery request)
+        {
+            var perimeter= _unitofwork.PerimeterRepository.FindById(request.PerimeterId);
+
+            if (perimeter == null)
+                throw new NullReferenceException("Perimeter with specified id does not exist!");
```

Comment:
Don`t throw NullReferenceExcpetion by yourself. i would like to reccomend ypu create you custom exception class.

Reply:
I left it for future